instrumentation/jmx-metrics/library/src/test/java/io/opentelemetry/instrumentation/jmx/engine/RuleParserTest.java

```diff
@@ -589,23 +604,33 @@ private static JmxConfig parseConf(String s) {
     InputStream is = new ByteArrayInputStream(s.getBytes(StandardCharsets.UTF_8));
     JmxConfig jmxConfig = parser.loadConfig(is);
     assertThat(jmxConfig).isNotNull();
+
+    // building metric definitions allows triggering validation rules
+    for (JmxRule rule : jmxConfig.getRules()) {
+      try {
+        rule.buildMetricDef();
+      } catch (Exception e) {
+        throw new IllegalStateException(e.getMessage(), e);
```

Comment:
[for reviewer] here throwing the original message allows to keep the assertion on message content the same.